themes/bootstrap3/templates/search/searchbox.phtml

```diff
@@ -81,135 +81,155 @@
-    <?php
-      $searchboxAttributes = [
-        'id' => 'searchForm_lookfor',
-        'class' => 'searchForm_lookfor form-control search-query',
-        'type' => 'search',
-        'name' => 'lookfor',
-        'value' => $this->lookfor,
-        'aria-label' => $this->translate('search_terms'),
-      ];
-      if ($placeholder = $this->searchbox()->getPlaceholderText($tabConfig['selected']['id'] ?? $this->searchClassId)) {
-        $searchboxAttributes['placeholder'] = $this->translate($placeholder);
+    <style>
+      .searchForm-inputs {
+          display: flex;
+          flex-wrap: wrap;
+          gap: 0.5rem;
       }
-      if ($this->searchbox()->autocompleteEnabled($this->searchClassId)) {
-        $searchboxAttributes['class'] .= " autocomplete searcher:{$this->searchClassId}"
-          . ($this->searchbox()->autocompleteAutoSubmit($this->searchClassId) ? ' ac-auto-submit' : '');
-        $searchboxAttributes['data-autocomplete-formatting-rules'] = $this->searchbox()->autocompleteFormattingRulesJson($this->searchClassId);
+
+      .searchForm-inputs .searchForm-query {
+          width: auto;
+          flex-grow: 1;
       }
-      if (!empty($keyboardLayouts)) {
-        $searchboxAttributes['class'] .= ' with-keyboard-selection';
+
+      @media screen and (min-width: 768px) {
+          .searchForm-inputs .searchForm-query {
+              flex-grow: 0;
+          }
       }
-    ?>
-    <div class="searchForm-query">
-      <input<?=$this->htmlAttributes($searchboxAttributes)?>>
-      <div id="searchForm_controls">
-        <button id="searchForm-reset" class="searchForm-reset hidden" type="reset" tabindex="-1" aria-label="<?=$this->transEscAttr('searchform_reset_button')?>"><?=$this->icon('ui-reset-search');?></button>
-        <?php if (!empty($keyboardLayouts)): ?>
-          <?php
-          $this->headScript()->appendFile('vendor/js.cookie.js');
-          $this->headScript()->appendFile('vendor/simple-keyboard/index.js');
-          $this->headScript()->appendFile('vendor/simple-keyboard-layouts/index.js');
-          $this->headLink()->appendStylesheet('vendor/simple-keyboard/index.css');
-          ?>
-          <div class="keyboard-selection dropdown">
-            <button class="btn btn-primary dropdown-toggle" type="button" id="keyboard-selection-button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
-              <?=$this->icon('keyboard-o') ?>
-              <?=$this->icon('caret-down') ?>
-            </button>
-            <ul class="dropdown-menu" aria-labelledby="keyboard-selection-button">
-              <li>
-                <a class="dropdown-item keyboard-selection-item" href="#" data-value="none"><?= $this->transEsc('None') ?></a>
-              </li>
-              <?php foreach ($keyboardLayouts as $keyboardLayout): ?>
+    </style>
```

Comment:
These styles are just an example and would be removed before merge.  The effect is to let the main search box grow on mobile widths to take up whatever space it has available, but to still fit the Find button on the same line if there is nothing else in between.  It's nice space saver.